whylog/log_reader/__init__.py

```diff
@@ -21,22 +23,35 @@ def get_causes_tree(self, front_input):
 
 
 class LogReader(AbstractLogReader):
+    NO_TEMP = frozendict()
+
     def __init__(self, config):
         self.config = config
 
-    def get_causes(self, front_input):
-        input_log_type = self.config.get_log_type(front_input.line_source)
+    def get_causes(self, front_input, tmp_assign_to_log_type=NO_TEMP):
+        input_line_source = front_input.line_source
+        input_log_type = self.config.get_log_type(input_line_source)
```

Comment:
`tmp_assign_to_log_type` should have precedence over what config returns (this is needed for a rather complicated usecase, where the user forces the log type selection for a given file which technically matches some other type)